utils/src/main/java/io/serverlessworkflow/utils/WorkflowUtils.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2020-Present The Serverless Workflow Specification Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.serverlessworkflow.utils;
+
+import io.serverlessworkflow.api.Workflow;
+import io.serverlessworkflow.api.interfaces.State;
+import io.serverlessworkflow.api.start.Start;
+import java.util.Objects;
+
+/** Provides common utility methods to provide most often needed answers from a workflow */
+public final class WorkflowUtils {
+
+  /**
+   * Gets State matching Start state name If start is not present returns first state Returns null
+   * otherwise
+   *
+   * @param workflow workflow
+   * @return {@code state} when present else returns {@code null}
+   */
+  public static State getStartingState(Workflow workflow) {
+    Objects.requireNonNull(workflow);
+    if (workflow.getStates() == null || workflow.getStates().isEmpty()) {
+      return null;
+    }
+
+    Start start = workflow.getStart();
+    if (start == null) {
+      return workflow.getStates().stream().findFirst().get();
```

Comment:
can just return the first element in the list? no need for stream here i think

Reply:
you are right, over looked it .